src/rustdoc_schema.graphql

```diff
@@ -694,6 +694,37 @@ type Constant implements Item & Importable & GlobalValue {
   docs: String
   attrs: [String!]!
   visibility_limit: String!
+  # properties for Constant
+  """
+  A type represented in the "raw" rustdoc JSON representation, see interface RawType.
+  Currently serialized as a serde_json String instead of a proper RawType
+  ```rust
+  //          	                	            	// rustdocs type field
+  const MIN   	:          usize	= 16       ;	// {"primitive":"usize"}
+  const LOG_AS	: &'static str  	= "batch"  ;	// {"borrowed_ref":{"lifetime":"'static","mutable":false,"type":{"primitive":"str"}}}
+  const YEAR  	:          Years	= Years(42);	// {"resolved_path":{"name":"Years","id":"0:3:1633","args":{"angle_bracketed":{"args":[],"bindings":[]}}}}
+  ```
+  """
+  type_: String
+  """
+  The expression/value of the constant, if any, as a Rust literal, expression, or `"_"`, as well as its literal flag. For example:
+  ```rust
+  //            	                	            	// is_literal	value      	expr
+  const MIN     	:          usize	= 16       ;	// true      	`"16usize"`	`16`
+  const MIN_SIZE	:          usize	= MIN      ;	// false     	`"16usize"`	"MIN", referring to the other constant's name
+  const LOG_AS  	: &'static str  	= "batch"  ;	// true      	None       	`"\"batch\""` including escaped quotes
+  const YEAR    	:          Years	= Years(42);	// false     	None       	"_"
+  const EXPR_2_2	:          i32  	= 2 + 2    ;	// false     	`"4i32"    	"_"
+  const FN_FIVE 	:          i32  	= five()   ;	// false     	`"5i32"    	"_"
+  const fn five() -> i32 { 5 } ;
+  struct Years(i32);
+  ```
+  If the constant is set to be equal to another constant, the expr holds the name of that other constant, while the value holds the value of that other constant.
+  If the constant is set by evaluating a `const` expression, such as `2 + 2` or a `const fn` call, rustdoc's current behavior is to show an expr of `"_"` (the value is evaluated)
+  """
```

Comment:
This comment is excellent! Thanks for looking into and documenting all the edge cases.

As it's currently positioned, it technically only applies to the `expr` property here so editor integrations won't show it for `value` or `is_literal` even though it's clearly applicable there. Any thoughts on how to tweak it to document the other properties as well?

I can see a couple of options:
- Move the docs to apply to the `Constant` type itself, then add docs on each property with a brief explanation + a mention that the type docs have a complete example covering all the edge cases.
- Mostly duplicate that doc comment to each property, emphasizing the specific property's values in each doc.

Reply:
Hard for me to decide without being able to see how the docs appear "in practice", how do I check that locally?

I've added a brief note that refers to the `expr`, but now that you mention editor integrations that might not be good enough as you'd need a per-property doc?

If so, then then your second suggestion is better, but then I don't think it's much of help to have a table of each 3 properties in the general doc, so this should also be split into only including the relevant value

If not and if the docs are laid out sequentialy like on docs.rs, then I think it's easier to have a single "comprehensive example" with a brief not telling the reader to look up